extensions/transfer/transfer-store-cosmos/src/main/java/com/microsoft/dagx/transfer/store/cosmos/CosmosTransferProcessStore.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ *  All rights reserved.
+ *
+ */
+
+package com.microsoft.dagx.transfer.store.cosmos;
+
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.CosmosStoredProcedure;
+import com.azure.cosmos.implementation.NotFoundException;
+import com.azure.cosmos.implementation.RequestRateTooLargeException;
+import com.azure.cosmos.models.*;
+import com.microsoft.dagx.spi.DagxException;
+import com.microsoft.dagx.spi.transfer.store.TransferProcessStore;
+import com.microsoft.dagx.spi.types.TypeManager;
+import com.microsoft.dagx.spi.types.domain.transfer.TransferProcess;
+import com.microsoft.dagx.transfer.store.cosmos.model.TransferProcessDocument;
+import net.jodah.failsafe.Fallback;
+import net.jodah.failsafe.RetryPolicy;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static net.jodah.failsafe.Failsafe.with;
+
+public class CosmosTransferProcessStore implements TransferProcessStore {
```

Comment:
Idea for the future: Maybe its possible to make the stores more generic. Maybe a cosmos db store implementation can be used to store TransferProcessItems or read/write data within a FlowController